File: EntertainmentNetworkServer/EntertainmentNetworkServer/src/main/java/isd/dp/ua/EntertainmentNetworkServer/Interfaces/IHallService.java
package isd.dp.ua.EntertainmentNetworkServer.Interfaces;

import java.math.BigDecimal;
import java.util.List;

import javax.jws.WebService;

import isd.dp.ua.EntertainmentNetworkServer.Models.*;
import isd.dp.ua.EntertainmentNetworkServer.ServiceMessages.*;

@WebService
public interface IHallService 
{
	void addHall(AddHallRequest transientInstance);

	Hall findHallById(BigDecimal id);
	
	public List<Hall> getHalls();
	
	Hall mergeHall(MergeHallRequest detachedInstance);
	
	void removeHall(BigDecimal id);
}

File: EntertainmentNetworkClient/EntertainmentNetwork.DAL/Models/Interfaces/IHallService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EntertainmentNetwork.DAL.Models.Interfaces
{
    public interface IHallService
    {
        Task AddHall(IHall hall);

        Task<IHall> FindHallById(decimal id);

        Task<List<IHall>> GetHalls();

        Task<IHall> MergeHall(IHall hall);

        Task RemoveHall(decimal id);
    }
}
